From 85eba69effc1b19ffb04e04c673ddc891d753dec Mon Sep 17 00:00:00 2001
From: Zhou Peng <zhoupeng@loongson.cn>
Date: Mon, 14 Sep 2020 16:54:45 +0800
Subject: [PATCH 1/3] Fix wrong vector ability display in --disable-asm mode.

If we configure this option :'--disable-asm ' in the device of Loongson
3A4000, when we use x264 for encoding, here shows an incorrect cpu
capability information,like:
--x264 [info]: using cpu capabilities: MSA

Change-Id: I3c0e2226aed053e1c2e25e592dd560a91aec7f0c
Signed-off-by: Zhou Peng <zhoupeng@loongson.cn>
---
 common/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/cpu.c b/common/cpu.c
index e52d3a2..321415f 100644
--- a/common/cpu.c
+++ b/common/cpu.c
@@ -407,7 +407,7 @@ uint32_t x264_cpu_detect( void )
     return X264_CPU_ARMV8 | X264_CPU_NEON;
 }
 
-#elif ARCH_MIPS
+#elif ARCH_MIPS && HAVE_MSA
 
 uint32_t x264_cpu_detect( void )
 {
-- 
2.1.0

